lib/foreman_tasks/engine.rb

```diff
@@ -108,7 +108,7 @@ class Engine < ::Rails::Engine
       ForemanTasks.dynflow.require!
       ::ForemanTasks.dynflow.config.on_init(false) do |world|
         world.middleware.use Actions::Middleware::KeepCurrentTaxonomies
-        world.middleware.use Actions::Middleware::KeepCurrentUser
+        world.middleware.use Actions::Middleware::KeepCurrentUser, :before => ::Dynflow::Middleware::Common::Transaction
```

Comment:
This should solve the referenced Katello issue, right? or am I missing somethin?

Reply:
Yes, together with the extended period when current user is set